bowling_game.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+$LOAD_PATH.push('Users/andrespacheco/Documents/andres-pacheco/bowling-game-kata-pacheco-andres')
+
+load 'frame.rb'
+load 'game_roll.rb'
+
+# bowling game logic class
+class BowlingGame
+  def initialize
+    @pins_down_collection = [10, 2, 3, 0, 0, 4, 6, 0, 0, 3, 4, 10, 10, 9, 3, 8, 1]
+  end
+
+  def start
+    @frames = []
+    game_roll = GameRoll.new
+    pins_down_collection_iterator(game_roll)
+    @frames
+  end
+
+  def pins_down_collection_iterator(game_roll)
+    @pins_down_collection.each do |pins_down|
+      game_roll.add(pins_down)
+      frame = Frame.new
+
+      if @frames.count == 10
+        strike_or_spare(frame, pins_down)
+      elsif game_roll.strike? && @frames.count <= 9
+        frame.add(pins_down)
+        @frames << frame
+        game_roll.reset
+      elsif game_roll.completed? && @frames.count <= 9
+        frame.add(game_roll.first_turn_pins_down)
+        frame.add(game_roll.second_turn_pins_down)
+        @frames << frame
+        game_roll.reset
+      end
+    end
+  end
+
+  def strike_or_spare(frame, pins_down)
+    if @frames.last.strike? || @frames.last.spare?
+      frame.add(pins_down)
+      @frames << frame
+    end
+  end
+
+  def score
+    score = 0
+    bonus = 0
+    
+    @frames.each.with_index do |frame, index|
+      score += frame.turns.sum
```

Comment:
I'd create two methods in a separate class that get those values or delegate frame to get all the points
`frame.final_score` for example